src/react-components/ory/user-logout-card.spec.tsx

```diff
@@ -0,0 +1,41 @@
+import { expect, test } from "@playwright/experimental-ct-react"
+import { Locator } from "@playwright/test"
+import { UserLogoutCard } from "./user-logout-card"
+
+let component: Locator
+
+test.beforeEach(async ({ mount }) => {
+  component = await mount(
+    <UserLogoutCard
+      csrfToken="CSRF token"
+      action="https://example.com/logout"
+      challenge="logout challenge"
+    />,
+  )
+})
+
+test("submits when user clicks 'Yes'", async () => {
+  const [request] = await Promise.all([
+    component.page().waitForRequest("https://example.com/logout"),
+    component.locator("button", { hasText: /yes/i }).click(),
+  ])
+
+  expect(request.method()).toBe("POST")
+  const data = request.postDataJSON()
+  expect(data).toHaveProperty("_csrf", "CSRF token")
+  expect(data).toHaveProperty("challenge", "logout challenge")
+  expect(data).toHaveProperty("submit", "Yes")
```

Comment:
We also have this helper for testing `UserAuthPage` which you could pass along a flow object to check if the fields exist through a fixture.

So you have a fixture which is just json data stored [here](https://github.com/ory/elements/blob/main/src/test/Fixtures.ts) and then you have the `UserAuthPage` which has some methods to extract and test the fixtures against the actual rendered page [here](https://github.com/ory/elements/blob/main/src/test/AuthPage.ts).

Reply:
This of course assumes you are using flow data.

I think we should have these helpers in one place since they are packaged into the `@ory/elements-test` package. https://github.com/ory/elements/blob/main/packages/test/package.json

This package will be used as a tool by examples in Ory as well as, hopefully, our customers for tests. 
The idea is that we write tests once, have common selectors and common flows packaged which can be extended by each individual test case for edge cases. 

Here is a preview of an idea for common login tests 
https://github.com/ory/elements/blob/13551747b388932302390d46a1b38ae22caef02f/src/test/login/index.ts

Used by this react example 
https://github.com/ory/elements/blob/13551747b388932302390d46a1b38ae22caef02f/examples/react-spa/e2e/login.spec.ts